testing/runner.ts

```diff
@@ -95,74 +93,110 @@ export async function getMatchingUrls(
 
   return matchingLocalUrls.concat(matchingRemoteUrls);
 }
+
+export interface RunTestModulesOptions extends RunOptions {
+  exclude?: string[];
+}
+
 /**
- * This function runs matching test files in `root` directory.
+ * Import the specified test modules and run their tests as a suite.
+ *
+ * Test modules are specified as an array of strings and can include local files
+ * or URLs.
  *
- * File matching and excluding supports glob syntax, ie. if encountered arg is
- * a glob it will be expanded using `glob` method from `fs` module.
+ * File matching and excluding support glob syntax - arguments recognized as
+ * globs will be expanded using `glob()` from the `fs` module.
  *
- * Note that your shell may expand globs for you:
- *    $ deno -A ./runner.ts **\/*_test.ts **\/test.ts
+ * Example:
  *
- * Expanding using `fs.glob`:
- *    $ deno -A ./runner.ts \*\*\/\*_test.ts \*\*\/test.ts
+ *       runTestModules(["**\/*_test.ts", "**\/test.ts"]);
  *
- *  `**\/*_test.ts` and `**\/test.ts"` are arguments that will be parsed and
- *  expanded as: [glob("**\/*_test.ts"), glob("**\/test.ts")]
+ * Any matched directory `<dir>` will expand to:
+ *   - `<dir>/**\/*_test.ts`
+ *   - `<dir>/**\/*_test.js`
+ *   - `<dir>/**\/test.ts`
+ *   - `<dir>/**\/test.js`
+ *
+ * So the above example is captured naturally by:
+ *
+ *       runTestModules(["."]);
  */
-// TODO: change return type to `Promise<void>` once, `runTests` is updated
-// to return boolean instead of exiting
-export async function main(root: string = cwd()): Promise<void> {
+// TODO: Change return type to `Promise<void>` once, `runTests` is updated
+// to return boolean instead of exiting.
+export async function runTestModules(
+  include: string[],
+  {
+    exclude = [],
+    parallel = false,
+    exitOnFail = false,
+    only = /[^\s]/,
+    skip = /^\s*$/,
+    disableLog = false
```

Comment:
Add `failIfNoMatchingFiles` option (or something along the lines)

Reply:
Yes, WIP.